main/src/modules/bodySchema/sensoryProcessor/src/sensoryProcessor.cpp

```diff
@@ -282,10 +283,14 @@ bool SensoryProcessor::getMultimodalData()
     Size subPixWinSize(10,10), winSize(31,31);
 
     ImageOf<PixelRgb> *yarpImage = imgPortIn.read();
-    IplImage *cvImage = cvCreateImage(cvSize(yarpImage->width(), yarpImage->height()), IPL_DEPTH_8U, 3);
-    cvCvtColor((IplImage*)yarpImage->getIplImage(), cvImage, CV_RGB2BGR);
+    ImageOf<PixelBgr> tmp; tmp.resize(*yarpImage);
```

Comment:
Hmm .. is `tmp` needed at all? Can't you just do
```
Mat cvImage=cvarrToMat((IplImage*)yarpImage->getIplImage());
cvtColor(cvImage,cvImage,CV_RGB2BGR);
```
According to http://docs.opencv.org/2.4/modules/imgproc/doc/miscellaneous_transformations.html in-place color conversion should be possible.